src/S3/S3MultiRegionClient.php

```diff
@@ -279,11 +301,26 @@ public function determineBucketRegionAsync($bucketName)
         /** @var S3ClientInterface $regionalClient */
         $regionalClient = $this->getClientFromPool();
         return $regionalClient->determineBucketRegionAsync($bucketName)
-            ->then(function ($region) use ($cacheKey) {
-                $this->cache->set($cacheKey, $region);
+            ->then(
+                function ($region) use ($cacheKey) {
+                    $this->cache->set($cacheKey, $region);
+
+                    return $region;
+                },
+                function (AwsException $e) use ($cacheKey) {
```

Comment:
I thought this was reachable but couldn't generate an exception that escaped to this level that would be caught here; cleaning up.